toolkit/tools/scheduler/schedulerutils/buildlist.go

```diff
@@ -9,14 +9,61 @@ import (
 	"os"
 	"path/filepath"
 
+	"github.com/juliangruber/go-intersect"
 	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagegen/configuration"
 	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagegen/installutils"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/exe"
 	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
 	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/pkggraph"
 	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/pkgjson"
 	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/sliceutils"
 )
 
+func ParseAndGeneratePackageList(dependencyGraph *pkggraph.PkgGraph, pkgsToBuild, pkgsToRebuild, pkgsToIgnore, imageConfig, baseDirPath string) (finalPackagesToBuild, packagesToRebuild, packagesToIgnore []*pkgjson.PackageVer, err error) {
+	// Generate the list of packages that need to be built.
+	// If none are requested then all packages will be built.
+	packagesToBuild, err := PackageNamesToBuiltPackages(exe.ParseListArgument(pkgsToBuild), dependencyGraph)
```

Comment:
>exe.ParseListArgument(pkgsToBuild) [](http://example.com/codeflow?start=53&length=34)

Let's do this conversion outside the function and make the function accept a list of strings instead. Or to look at this another way: let's use natural language data structures inside helper functions and avoid forcing them to know the format of the input for a given tool. They shouldn't need to have that responsibility.

Reply:
Done